src/components/KMethodBadge/KMethodBadge.vue

```diff
@@ -0,0 +1,167 @@
+<template>
+  <KBadge
+    :background-color="backgroundColor || methodBadgeColors[method].backgroundColor"
+    :border-color="borderColor || methodBadgeColors[method].borderColor"
+    class="k-method-badge"
+    :class="[ `method-${method}`, isToggle ? 'k-method-toggle' : '']"
+    :color="color || methodBadgeColors[method].color"
+    :is-bordered="isToggle"
+    max-width="auto"
+    :shape="shape === 'rectangular' ? 'rectangular' : 'rounded'"
+    v-on="isToggle ? { click: toggleValue } : {}"
+  >
+    <span v-if="!isToggle">{{ methodLabel }}</span>
+    <KInputSwitch
+      v-if="isToggle"
+      v-model="modelValue"
+      :label="methodLabel"
+      label-position="left"
+    />
+  </KBadge>
+</template>
+
+<script lang="ts">
+import { PropType, computed, ref, watch } from 'vue'
+import { MethodsArray, ShapesArray } from '@/types'
+import type { Method, MethodShape, MethodBadgeColors } from '@/types'
+</script>
+
+<script setup lang="ts">
+const props = defineProps({
+  method: {
+    type: String as PropType<Method>,
+    required: true,
+    validator: (value: Method): boolean => {
+      return MethodsArray.includes(value as Method)
+    },
+    default: 'custom',
+  },
+  shape: {
+    type: String as PropType<MethodShape>,
+    required: false,
+    validator: (value: MethodShape): boolean => {
+      return ShapesArray.includes(value as MethodShape)
+    },
+    default: 'rounded',
+  },
+  label: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  color: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  backgroundColor: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  borderColor: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  isToggle: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  value: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+})
+
+const emit = defineEmits(['update:modelValue'])
+
+const modelValue = ref<boolean>(props.value)
```

Comment:
It seems weird to set a ref to a prop

Reply:
In my opinion very efficient way to set it right on component setup. Maybe needs a watcher though, to watch for prop changes - just added it.